template.go

```diff
@@ -73,11 +73,22 @@ func NewTemplater(fileSystem afero.Fs, templateRootPath, outputRootPath string,
 // Execute is responsible of parsing and executing the given template, using the passed data context.
 // If execution is successful, the template is written to it's desired target location.
 // If allowNoValue is true, the template is rendered even if it contains variables which are not defined.
-func (t *Templater) Execute(template *TemplateFile, data any, allowNoValue bool) error {
-	return t.execute(template, data, template.Path, allowNoValue)
+func (t *Templater) Execute(template *TemplateFile, data any, allowNoValue bool, renameConfig []RenameConfig) error {
+	// if a renameConfig exists, rename possible files accordingly
+	if renameConfig != nil {
+		fmt.Println("we got a renameConfig")
+		fmt.Println(renameConfig)
+		for _, rename := range renameConfig {
+			if strings.EqualFold(template.Path, rename.InputPath) {
+				fmt.Println("new output:", rename.Filename)
+				return t.execute(template, data, rename.Filename, allowNoValue, nil)
+			}
+		}
+	}
+	return t.execute(template, data, template.Path, allowNoValue, nil)
 }
 
-func (t *Templater) execute(template *TemplateFile, data any, targetPath string, allowNoValue bool) error {
+func (t *Templater) execute(template *TemplateFile, data any, targetPath string, allowNoValue bool, renameConfig []RenameConfig) error {
```

Comment:
Is adding `renameConfig` still required? The targetPath should be sufficient and it seems that you're not using it.

Reply:
nope, you're right. That was the first draft, will remove - thanks for the headsup!